node/core/provisioner/src/lib.rs

```diff
@@ -324,32 +325,30 @@ fn select_availability_bitfields(
 	cores: &[CoreState],
 	bitfields: &[SignedAvailabilityBitfield],
 ) -> Vec<SignedAvailabilityBitfield> {
-	let mut bitfield_per_core: Vec<Option<SignedAvailabilityBitfield>> = vec![None; cores.len()];
-	let mut seen_validators = HashSet::new();
+	let mut selected: BTreeMap<ValidatorIndex, SignedAvailabilityBitfield> = BTreeMap::new();
 
-	for mut bitfield in bitfields.iter().cloned() {
-		// If we have seen the validator already, ignore it.
-		if !seen_validators.insert(bitfield.validator_index()) {
-			continue;
+	'a:
+	for bitfield in bitfields.iter().cloned() {
+		if bitfield.payload().0.len() != cores.len() {
+			continue
 		}
 
-		for (idx, _) in cores.iter().enumerate().filter(|v| v.1.is_occupied()) {
+		let is_better = selected.get(&bitfield.validator_index())
```

Comment:
If we accepted the other bitfield and the new one has more ones it must mean that it is invalid? Because it would have a bit set for a free core

Reply:
why? we don't require that all occupied cores have the bit set.

Reply:
🤦🤦🤦